### PR TITLE
Reverse-Many-To-Many-Object-Relation: Allowed Many to One relation

### PR DESCRIPTION
**Pull request by gauravsharma-pimcore**
## Changes in this pull request  
This pull request allows the "Many To One relation" type field in the "Reverse Many to Many object relation" field. As we have use case of many to one relationship in classes and want to show linked object in other classes through "Reverse Many to Many object relation".

### CODE REVIEW

**Comment by brusch** [comment]
@gauravsharma-pimcore  Thanks for your PR, can you please just sign our CLA? Thanks a lot!

**Comment by dvesh3** [comment]
@gauravsharma-pimcore thanks a lot for your PR. really nice to see that with one liner you introduced a new feature 😉 I believe we need more refactoring here as data type name is not valid anymore "Reverse-**Many-To-Many**-Object-Relation" maybe we should rename it to just "Reverse-Object-Relation". wdyt?

**Comment by gauravsharma-pimcore** [comment]
> @gauravsharma-pimcore Thanks for your PR, can you please just sign our CLA? Thanks a lot!

@brusch I have signed the CLA.

**Comment by gauravsharma-pimcore** [comment]
> @gauravsharma-pimcore thanks a lot for your PR. really nice to see that with one liner you introduced a new feature  I believe we need more refactoring here as data type name is not valid anymore "Reverse-**Many-To-Many**-Object-Relation" maybe we should rename it to just "Reverse-Object-Relation". wdyt?

@dvesh3 : I am doing the refactoring. Thanks.

**Comment by brusch** [comment]
@gauravsharma-pimcore CLA is still not signed with the correct email. Please ensure you have configured the email address your using for git also on GitHub. You can see the email used for git here: https://patch-diff.githubusercontent.com/raw/pimcore/pimcore/pull/7963.patch

**Comment by gauravsharma-pimcore** [comment]
> @gauravsharma-pimcore CLA is still not signed with the correct email. Please ensure you have configured the email address your using for git also on GitHub. You can see the email used for git here: https://patch-diff.githubusercontent.com/raw/pimcore/pimcore/pull/7963.patch

@brusch I have updated the email address and I think CLA is signed now. Please confirm.

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseObjectRelation.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ pimcore.object.classes.data.reverseManyToManyObjectRelation = Class.create(pimco
                     rootProperty: "availableFields"
                 },
                 extraParams: {
-                    types: 'manyToManyObjectRelation',
+                    types: 'manyToManyObjectRelation,manyToOneRelation',
```
Comment (blankse) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseObjectRelation.js:125]: Is it also possible with manyToManyRelation?

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 2]
```diff
@@ -189,7 +189,7 @@ pimcore:
 ```
 
 - Using static methods for [dynamic text labels](../../05_Objects/01_Object_Classes/03_Layout_Elements/01_Dynamic_Text_Labels.md) is now deprecated, use services instead.
-- Removed method `\Pimcore\Model\DataObject\ClassDefinition\Data\Relations\AbstractRelations::isRemoteOwner()`, as this method was only used for `ReverseManyToManyObjectRelation` internal check are now made using `instanceof` 
+- Removed method `\Pimcore\Model\DataObject\ClassDefinition\Data\Relations\AbstractRelations::isRemoteOwner()`, as this method was only used for `ReverseObjectRelation` internal check are now made using `instanceof`
```
Comment (dvesh3) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:192]: No need to rename here...as it stays `ReverseManyToManyObjectRelation` on 6.5.0

File: bundles/CoreBundle/Resources/translations/cs.extended.json [hunk 3]
```diff
@@ -687,7 +687,7 @@
     "custom_login_background_image": "Custom Login Background Image",
     "show_charcount": "Show Character Count",
     "max_length": "Max Length",
-    "reverse_many_to_many_object_relation": "Reverse Many-To-Many Object Relation",
+    "reverse_object_relation": "Reverse Object Relation",
```
Comment (dvesh3) [inline bundles/CoreBundle/Resources/translations/cs.extended.json:690]: please revert all translation files except en.extended.json. Translations are managed in POEditor: https://poeditor.com/join/project/VWmZyvFVMH

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md (hunk 2 above)
Comment (gauravsharma-pimcore) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: Right. Reverted the file.

File: bundles/CoreBundle/Resources/translations/cs.extended.json (hunk 3 above)
Comment (gauravsharma-pimcore) [reply bundles/CoreBundle/Resources/translations/cs.extended.json]: Reverted the files.

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseObjectRelation.js (hunk 1 above)
Comment (gauravsharma-pimcore) [reply bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseObjectRelation.js:125]: Yes. It is also possible with manyToManyRelation but as we use manyToManyObjectRelation for objects that's why I didn't include manyToManyRelation.

**Comment by dvesh3** [comment]
@gauravsharma-pimcore thanks a lot for your changes. 

We have decided that these changes should go into 6.9 with deprecations. and then we remove deprecated stuff on Pimcore 10.
so please:
1. Rebase this PR on branch 6.9

2. Add more changes: we also need to include BC layer, so that existing implementations and bundles should not break.
  -  class alias on model: /models/DataObject/ClassDefinition/Data/ReverseObjectRelation.php
  ```php
  class_alias(ReverseObjectRelation::class, 'Pimcore\Model\DataObject\ClassDefinition\Data\Nonownerobjects');
  class_alias(ReverseObjectRelation::class, 'Pimcore\Model\DataObject\ClassDefinition\Data\ReverseManyToManyObjectRelation');
  ```
  -  deprecated class like https://github.com/pimcore/pimcore/blob/6.9/models/DataObject/ClassDefinition/Data/Nonownerobjects.php
  - class alias on js: https://github.com/pimcore/pimcore/blob/7d31cd63664c723bb5c0d756e4595b2ab66bde24/bundles/AdminBundle/Resources/public/js/pimcore/object/tags/reverseManyToManyObjectRelation.js#L426
      ```js 
      // @TODO BC layer, to be removed in Pimcore 10
      pimcore.object.tags.nonownerobjects = pimcore.object.tags.reverseObjectRelation;
      pimcore.object.tags.reverseManyToManyObjectRelation = pimcore.object.tags.reverseObjectRelation;
      ```
       
      and
 https://github.com/pimcore/pimcore/blob/ae49ec7ce30de1db8f9b69b1dfad87d0f82f33b9/bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseManyToManyObjectRelation.js#L181
       ```js
      // @TODO BC layer, to be removed in Pimcore 10
      pimcore.object.classes.data.nonownerobjects = pimcore.object.classes.data.reverseObjectRelation;
      pimcore.object.classes.data.reverseManyToManyObjectRelation = pimcore.object.classes.data.reverseObjectRelation;
      ```
      
Thanks for your co-operation🙏

**Comment by gauravsharma-pimcore** [comment]
> @gauravsharma-pimcore thanks a lot for your changes.
> 
> We have decided that these changes should go into 6.9 with deprecations. and then we remove deprecated stuff on Pimcore 10.
> so please:
> 
> 1. Rebase this PR on branch 6.9
> 2. Add more changes: we also need to include BC layer, so that existing implementations and bundles should not break.
> 
> * class alias on model: /models/DataObject/ClassDefinition/Data/ReverseObjectRelation.php
> 
> ```
> class_alias(ReverseObjectRelation::class, 'Pimcore\Model\DataObject\ClassDefinition\Data\Nonownerobjects');
> class_alias(ReverseObjectRelation::class, 'Pimcore\Model\DataObject\ClassDefinition\Data\ReverseManyToManyObjectRelation');
> ```
> 
> * deprecated class like https://github.com/pimcore/pimcore/blob/6.9/models/DataObject/ClassDefinition/Data/Nonownerobjects.php
> * class alias on js: https://github.com/pimcore/pimcore/blob/7d31cd63664c723bb5c0d756e4595b2ab66bde24/bundles/AdminBundle/Resources/public/js/pimcore/object/tags/reverseManyToManyObjectRelation.js#L426
>   
>   ```js
>   // @TODO BC layer, to be removed in Pimcore 10
>   pimcore.object.tags.nonownerobjects = pimcore.object.tags.reverseObjectRelation;
>   pimcore.object.tags.reverseManyToManyObjectRelation = pimcore.object.tags.reverseObjectRelation;
>   ```
>   
>   
>   and
>   https://github.com/pimcore/pimcore/blob/ae49ec7ce30de1db8f9b69b1dfad87d0f82f33b9/bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseManyToManyObjectRelation.js#L181
>   
>   ```js
>   // @TODO BC layer, to be removed in Pimcore 10
>   pimcore.object.classes.data.nonownerobjects = pimcore.object.classes.data.reverseObjectRelation;
>   pimcore.object.classes.data.reverseManyToManyObjectRelation = pimcore.object.classes.data.reverseObjectRelation;
>   ```
> 
> Thanks for your co-operation

Ok. will do that.

**Comment by gauravsharma-pimcore** [comment]
> > @gauravsharma-pimcore thanks a lot for your changes.
> > We have decided that these changes should go into 6.9 with deprecations. and then we remove deprecated stuff on Pimcore 10.
> > so please:
> > 
> > 1. Rebase this PR on branch 6.9
> > 2. Add more changes: we also need to include BC layer, so that existing implementations and bundles should not break.
> > 
> > 
> > * class alias on model: /models/DataObject/ClassDefinition/Data/ReverseObjectRelation.php
> > 
> > ```
> > class_alias(ReverseObjectRelation::class, 'Pimcore\Model\DataObject\ClassDefinition\Data\Nonownerobjects');
> > class_alias(ReverseObjectRelation::class, 'Pimcore\Model\DataObject\ClassDefinition\Data\ReverseManyToManyObjectRelation');
> > ```
> > 
> > 
> > 
> > * deprecated class like https://github.com/pimcore/pimcore/blob/6.9/models/DataObject/ClassDefinition/Data/Nonownerobjects.php
> > * class alias on js: https://github.com/pimcore/pimcore/blob/7d31cd63664c723bb5c0d756e4595b2ab66bde24/bundles/AdminBundle/Resources/public/js/pimcore/object/tags/reverseManyToManyObjectRelation.js#L426
> >   
> >   ```js
> >   // @TODO BC layer, to be removed in Pimcore 10
> >   pimcore.object.tags.nonownerobjects = pimcore.object.tags.reverseObjectRelation;
> >   pimcore.object.tags.reverseManyToManyObjectRelation = pimcore.object.tags.reverseObjectRelation;
> >   ```
> >   
> >   
> >   and
> >   https://github.com/pimcore/pimcore/blob/ae49ec7ce30de1db8f9b69b1dfad87d0f82f33b9/bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/reverseManyToManyObjectRelation.js#L181
> >   
> >   ```js
> >   // @TODO BC layer, to be removed in Pimcore 10
> >   pimcore.object.classes.data.nonownerobjects = pimcore.object.classes.data.reverseObjectRelation;
> >   pimcore.object.classes.data.reverseManyToManyObjectRelation = pimcore.object.classes.data.reverseObjectRelation;
> >   ```
> > 
> > Thanks for your co-operation
> 
> Ok. will do that.

I have created a new PR for the same as I was getting too many conflicts while rebasing.
PR: https://github.com/pimcore/pimcore/pull/8037

**Pull request closed without merging**